### PR TITLE
feat(YomuMangás): add presence

### PR DESCRIPTION
**Pull request by Vitu2002**
## Description 
Adding a new presence of a portuguese manga reader called "YomuMangás".
 - [Website](https://yomumangas.com)

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

 - [Settings](https://i.imgur.com/JJPJcZh.png)
   - [Images + User](https://i.imgur.com/Q94HsUD.jpg)
   - [All Disabled | Image | Image + Manga](https://i.imgur.com/apmVB6w.jpg)
   - [Image + Manga + Chapter](https://i.imgur.com/fRiQSOy.jpg)
   - [NSFW Manga | Privacy Mode](https://i.imgur.com/nhpU97l.jpg)
  
</details>

### CODE REVIEW

File: websites/Y/YomuMangás/metadata.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			"icon": "fad fa-exclamation-triangle",
+			"value": false
+		},
+		{
+			"id": "privacy",
+			"title": "Modo Privado",
+			"icon": "fad fa-user-secret",
+			"value": false
+		}
+	]
```
Comment (EncryptedDev) [inline websites/Y/YomuMangás/metadata.json:71]: What do each one of these do please?

File: websites/Y/YomuMangás/presence.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+				presence.getSetting<boolean>("chapter"),
+				presence.getSetting<boolean>("user"),
+				presence.getSetting<boolean>("timestamp"),
+				presence.getSetting<boolean>("adult"),
+				presence.getSetting<boolean>("privacy"),
+			]);
+
+	if (privacy) {
+		presenceData.details = "Criando um novo mundo!";
+		presenceData.state = "yomumangas.com";
```
Comment (EncryptedDev) [inline websites/Y/YomuMangás/presence.ts:20]: This is not needed

File: websites/Y/YomuMangás/presence.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+			chapterNumber = document.querySelector(
+				"#premid-chapter-number"
+			)?.textContent,
+			chapterTitle = document.querySelector(
+				"#premid-chapter-title"
+			)?.textContent;
+
+		if (images && !isAdult) {
+			presenceData.largeImageKey = (
+				document.querySelector("#premid-manga-image") as HTMLAnchorElement
```
Comment (EncryptedDev) [inline websites/Y/YomuMangás/presence.ts]:
```suggestion
				document.querySelector<HTMLAnchorElement>("#premid-manga-image")
```

File: websites/Y/YomuMangás/presence.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+				chapterTitle ? ` - ${chapterTitle}` : ""
+			}`;
+		}
+		if (!chapter && isReader) presenceData.state = "Lendo Capítulo";
+		if (timestamp) presenceData.startTimestamp = readingTimestamp;
+		if (manga && ((isAdult && adult) || !isAdult) && !(chapter && isReader)) {
+			presenceData.buttons = [
+				{
+					label: "Ver Mangá",
+					url: `https://yomumangas.com/manga/${mangaId}/${mangaSlug}`,
```
Comment (EncryptedDev) [inline websites/Y/YomuMangás/presence.ts:57]: Can't you use `href` for these three buttons?

**Review by EncryptedDev — COMMENTED** [review]
Hey! I went ahead to fix some preference-related stuff as shown in the latest commit so things are dealt with faster. Please also do respond to the following comments

File: websites/Y/YomuMangás/metadata.json (hunk 1 above)
Comment (Vitu2002) [reply websites/Y/YomuMangás/metadata.json:71]: Mostrar Images/Show Images, this option when enabled show in the large image the current manga cover (if not adult content) and the avatar from the user that you are looking

Mostrar Mangá/Show Manga, this option when enabled show the current manga when you are reading or on the manga page, if disabled, "Lendo um Mangá"/"Reading a Manga"

Mostrar Capítulo/Show Chapter, has the same function of the "Show Manga", but with the reading chapter

Mostrar Timestamp/Show Timestamp just show the how much time the user are in the website

Mostrar NSFW/Show NSFW, when enabled it's show only the name of the current manga that you are reading

Modo Privado/Privacy Mode, this options make the code ignore all pages and only show that the user are in the website.

File: websites/Y/YomuMangás/presence.ts (hunk 4 above)
Comment (Vitu2002) [reply websites/Y/YomuMangás/presence.ts:57]: Yeah, Then I'll change the website code to pass a valid href and remove the consts "mangaId" and "mangaSlug"

File: websites/Y/YomuMangás/presence.ts (hunk 2 above)
Comment (Vitu2002) [reply websites/Y/YomuMangás/presence.ts:20]: Removed

File: websites/Y/YomuMangás/presence.ts (hunk 3 above)
Comment (Vitu2002) [reply websites/Y/YomuMangás/presence.ts]: Changed

File: websites/Y/YomuMangás/metadata.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	"color": "#121212",
+	"category": "anime",
+	"tags": [
+		"anime",
+		"manga",
+		"doushinji",
+		"geek",
+		"nsfw"
+	],
+	"settings": [
```
Comment (DarkVillager) [inline websites/Y/YomuMangás/metadata.json:28]: Let's put privacy at the top and if you want it to stop showing images etc when privacy is on check [these settings](https://github.com/PreMiD/Presences/blob/main/websites/Y/YouTube/metadata.json#L99)

File: websites/Y/YomuMangás/presence.ts [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+	else if (pathname.includes("/minigames"))
+		presenceData.details = "Página de Minigames";
+	else if (pathname.includes("/donators"))
+		presenceData.details = "Página de Doadores";
+	else if (pathname.includes("/staffs"))
+		presenceData.details = "Página da Equipe";
+	else if (pathname.includes("/partners"))
+		presenceData.details = "Página de Parceiros";
+	else if (pathname.includes("/settings"))
+		presenceData.details = "Página de Configurações";
```
Comment (DarkVillager) [inline websites/Y/YomuMangás/presence.ts:119]: > Add the static pages to a Record, shown in the example here
> 
> https://github.com/PreMiD/Presences/blob/cf3631226067d77ada628adbf77d7d5e5e2ca445/websites/T/TruckersMP/presence.ts#L15-L41

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/Y/YomuMangás/metadata.json (hunk 5 above)
Comment (DarkVillager) [inline websites/Y/YomuMangás/metadata.json:28]: Let's put privacy at the top and if you want it to stop showing images etc when privacy is on check these settings

https://github.com/PreMiD/Presences/blob/main/websites/Y/YouTube/metadata.json#L96-L105

Comment (DarkVillager) [inline websites/Y/YomuMangás/metadata.json:28]: Let's put privacy at the top and if you want it to stop showing images etc when privacy is on check these settings

https://github.com/PreMiD/Presences/blob/a353e7d295ceebd90fc4a9b4e34b2efe12e5802c/websites/Y/YouTube/metadata.json#L96-L105

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/Y/YomuMangás/metadata.json (hunk 5 above)
Comment (Vitu2002) [reply websites/Y/YomuMangás/metadata.json:28]: I put the privacy on the top, but i think that's not necessary use the if in the metadata, the precense already stop showing the pages data when enabled

File: websites/Y/YomuMangás/presence.ts (hunk 6 above)
Comment (Vitu2002) [reply websites/Y/YomuMangás/presence.ts:119]: Done, now the static presence pages are a Record

File: websites/Y/YomuMangás/metadata.json (hunk 5 above)
Comment (DarkVillager) [reply websites/Y/YomuMangás/metadata.json:28]: bruh, but we prefer the metadata over the presence.

Comment (Vitu2002) [reply websites/Y/YomuMangás/metadata.json:28]: Hmm, I think there is no difference in this case, but I'll add in the metadata then

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**